### PR TITLE
Fix for #16 (loadProducts does not return any error when there is no network)

### PR DESCRIPTION
**Pull request by wasedaigo**
In case of no network, loadProducts should return an error instead of returning nothing.

### CODE REVIEW

**Comment by chirag04** [comment]
looks good. @agolovenko can you confirm this works for you?

**Comment by BodySplash** [comment]
Hi, I was about to do the very same PR. Any updates on when it should be merged ?

**Comment by chirag04** [comment]
i'm worried that will fail when restore purchase request fails.

**Comment by BodySplash** [comment]
What do you fear exactly ? 

I just noticed also that this PR does not remove the request from _callbacks.

**Comment by chirag04** [comment]
ok. sorry i got confused. restore purchases is not using `SKRequest`. so this is fine.

File: InAppUtils/InAppUtils.m [hunk 1]
```diff
@@ -204,6 +204,13 @@ - (void)productsRequest:(SKProductsRequest *)request
     }
 }
 
+- (void)request:(SKRequest *)request didFailWithError:(NSError *)error
```
Comment (chirag04) [inline InAppUtils/InAppUtils.m]: add a note that this used when loadProducts fails

File: InAppUtils/InAppUtils.m [hunk 2]
```diff
@@ -204,6 +204,13 @@ - (void)productsRequest:(SKProductsRequest *)request
...
+{
+    NSString *key = RCTKeyForInstance(request);
+    RCTResponseSenderBlock callback = _callbacks[key];
+    callback(@[RCTJSErrorFromNSError(error)]);
```
Comment (chirag04) [inline InAppUtils/InAppUtils.m]: remove the key from _callbacks

**Pull request closed without merging**